pkg/inference/models/manager.go

```diff
@@ -163,6 +168,17 @@ func (m *Manager) handleCreateModel(w http.ResponseWriter, r *http.Request) {
 
 	// Pull the model. In the future, we may support additional operations here
 	// besides pulling (such as model building).
+	haveMem, err := m.memoryEstimator.HaveSufficientMemoryForModel(r.Context(), request.From, nil)
+	if err != nil {
+		m.log.Warnf("Failed to calculate memory required for model %q: %s", request.From, err)
+		http.Error(w, "Could not calculate runtime memory requirement for model", http.StatusInternalServerError)
+		return
+	}
+	if !haveMem {
+		m.log.Warnf("Runtime memory requirement for model %q exceeds total system memory", request.From)
+		http.Error(w, "Runtime memory requirement for model exceeds total system memory", http.StatusInsufficientStorage)
+		return
```

Comment:
I was considering wether there were any cases where a user may want to pull a model on a system where there is insufficient memory to run it.

Some edge cases I thought of:
1. transferring between registries where you might pull, tag, and then push without ever running the model
2. maybe some future packaging use cases where you are pulling a model and then extending/modifying it?? but in these cases I would actually prefer if we accomplish this without the need to pull remote layers

If we think there are any potential uses for pulling a model to a system where it can't run we may want to add a `force` param to explicitly ignore the check.